src/lib/core/say.c

```diff
@@ -1364,11 +1435,20 @@ log_vsay(struct log *log, int level, bool check_level, const char *module,
 	if (check_level && level > log->level)
 		goto out;
 
+	if (is_initialized == 0) {
+		log->isatty = isatty(log->fd);
+		log->isatty_stderr = isatty(STDERR_FILENO);
+		is_initialized = 1;
+	}
+
```

Comment:
Why do we need this check and this isatty call here? IINM, the logger was already created via log_create, hence, should be fully initialized.

Reply:
added a comment to code